batavia/VirtualMachine.js

```diff
@@ -755,95 +755,161 @@ VirtualMachine.prototype.create_traceback = function() {
+                    'next_pos': pos
+                }
+                continue
             }
-            unpacked_code[opcode_start_pos + 2] = {
-                'opoffset': opcode_start_pos + 2,
-                'opcode': dis.NOP,
-                'op_method': this.dispatch_table[dis.NOP],
-                'args': [],
-                'next_pos': pos
+
+            var intArg = code.co_code.val[pos++] | extra
+            extra = 0
+
+            if (opcode >= dis.HAVE_ARGUMENT) {
+                if (opcode in dis.hasconst) {
+                    args = [code.co_consts[intArg]]
+                } else if (opcode in dis.hasfree) {
+                    if (intArg < code.co_cellvars.length) {
+                        args = [code.co_cellvars[intArg]]
+                    } else {
+                        var var_idx = intArg - code.co_cellvars.length
+                        args = [code.co_freevars[var_idx]]
+                    }
+                } else if (opcode in dis.hasname) {
+                    args = [code.co_names[intArg]]
+                } else if (opcode in dis.hasjrel) {
+                    args = [pos + intArg]
+                } else if (opcode in dis.hasjabs) {
+                    args = [intArg]
+                } else if (opcode in dis.haslocal) {
+                    args = [code.co_varnames[intArg]]
+                } else {
+                    args = [intArg]
+                }
             }
-            unpacked_code[opcode_start_pos + 3] = {
-                'opoffset': opcode_start_pos + 3,
-                'opcode': dis.NOP,
-                'op_method': this.dispatch_table[dis.NOP],
-                'args': [],
+
+            unpacked_code[opcode_start_pos] = {
+                'opoffset': opcode_start_pos,
+                'opcode': opcode,
+                'op_method': this.dispatch_table[opcode],
+                'args': args,
                 'next_pos': pos
             }
-            continue
         }
 
-        if (opcode < dis.HAVE_ARGUMENT) {
-            args = []
-        } else {
-            lo = code.co_code.val[pos++]
-            hi = code.co_code.val[pos++]
-            var intArg = lo | (hi << 8) | extra
-            extra = 0 // use extended arg if present
-
-            if (opcode in dis.hasconst) {
-                args = [code.co_consts[intArg]]
-            } else if (opcode in dis.hasfree) {
-                if (intArg < code.co_cellvars.length) {
-                    args = [code.co_cellvars[intArg]]
-                } else {
-                    var var_idx = intArg - code.co_cellvars.length
-                    args = [code.co_freevars[var_idx]]
+        code.co_unpacked_code = unpacked_code
+    } else {
+        // Until 3.6 Python had variable width opcodes
+
+        var pos = 0
+        var unpacked_code = []
+        var args
+        var extra = 0
+        var lo
+        var hi
+
+        while (pos < code.co_code.val.length) {
+            var opcode_start_pos = pos
```

Comment:
At column 13: [(no-redeclare) 'extra' is already defined.](http://eslint.org/docs/rules/no-redeclare)